src/vm/method.cpp

```diff
@@ -4769,14 +4814,32 @@ Precode* MethodDesc::GetOrCreatePrecode()
     }
 
     PTR_PCODE pSlot = GetAddrOfSlot();
-    PCODE tempEntry = GetTemporaryEntryPoint();
+    PCODE pExpected;
+    if (IsZapped())
+    {
+        // If the method is zapped, then the only way it may not have a stable entry point yet
+        // is that it has a precode. In this case it's not marked as having precode through HasPrecode
+        // instead the precode acts as a temporary entry point.
+        pExpected = *pSlot;
+#ifdef FEATURE_PREJIT
+        Module * pZapModule = GetZapModule();
```

Comment:
I think there is a race if other thread calls GetOrCreatePrecode at the same time - the slot is not going to point to zapped precode anymore.